exercises/change/example.go

```diff
@@ -0,0 +1,191 @@
+		for b := 0; b < n; b++ {
+			// smaller b is larger coin
+			if (m & (1 << uint(b))) != 0 {
+				sum += coins[n-1-b]
+				if sum > target {
+					// Already sum exceeds target for this combination,
+					// so consider it no more.
+					continue comboLoop
+				}
+				c = append(c, coins[n-1-b])
+			}
+		}
+		// Now the combination of coins in c can be considered.
+		// The sum of coins in c doesn't exceed target.
+		// When the sum matches the target, those coins are a solution.
+		if sum == target {
+			updateSolution(c, s, &bestsize)
+			continue
+		}
+		// Recursively search for a best solution.
+		partialSolution(nil, c, target, s, &bestsize)
+	}
+}
+
+// updateSolution updates solution with newsolution if no prior solution
+// or when length of newsolution is better.
+func updateSolution(newsolution []int, solution *[]int, bestsize *int) {
+	if len(*solution) == 0 || len(newsolution) < len(*solution) {
+		*bestsize = len(newsolution)
+		*solution = make([]int, len(newsolution))
+		copy(*solution, newsolution)
+		sort.Ints(*solution)
+	}
+}
+
+// partialSolution takes a partial solution, and coin list c looking to meet target
+// while updating solution s to include partial plus the coins meeting target.
+func partialSolution(partial []int, c []int, target int, s *[]int, bestsize *int) {
+	// We want to consider coins in c in largest to smallest order.
+	sort.Sort(sort.Reverse(sort.IntSlice(c)))
+
+	for e := range c {
+		// Find n, which is highest number of c[e] that could be used in change.
+		n := target / c[e]
+
+		// Try n, n-1 ... 1
+		for k := n; k >= 1; k-- {
+			if (k * c[e]) == target {
+				// k coins of c[e] completes a solution.
+				if (len(partial) + k) < *bestsize {
+					list := append(partial, nCoins(k, c[e])...)
+					updateSolution(list, s, bestsize)
+				}
+				// No need to consider smaller k for c[e], so exit this loop.
+				break
+			}
+			if len(c) > 1 {
+				// k coins of *c[e] might be potential partial solution.
+				newtarget := target - (k * c[e])
+				// Use a heuristic estimate of whether a partial solution
+				// might be worth it: if the larget coin in c not yet used
+				// in an already found solution allows the new solution
+				// to be smaller, then it would be worth pursuing.
+
+				// Find the largest coin in c not yet used in *s solution
+				// omitting c[e], since it is used in this partial solution.
+				largest := largestNotUsed(c, *s, c[e])
+				// It will take at least the partial, plus k coins plus 1 more.
+				if (len(partial) + k + 1) < *bestsize {
+					// Calculate the minimum potential size of a solution
+					// which doesn't include coin c[e].
+					potentialSize := len(partial) + k + (newtarget / largest)
+					if (newtarget % largest) != 0 {
+						potentialSize++
+					}
+					if potentialSize < *bestsize {
+						// Put k instances of c[e] in a new partial solution.
+						newPartial := append(partial, nCoins(k, c[e])...)
+						d := make([]int, len(c))
+						copy(d, c)
```

Comment:
optional: I'm not sure if this makes things any better, but you could just copy `c[:e]` instead of all of `c` since all elements in positions later than `e` are being overwritten anyway.

https://play.golang.org/p/TyAGEkq5L8